charts/DiscreteBarTransform.ts

```diff
@@ -4,27 +4,21 @@ import {
     isEmpty,
     sortBy,
     orderBy,
-    max,
     values,
     flatten,
-    uniq
+    uniq,
+    sortedUniq
 } from "./Util"
-import { ChartConfig } from "./ChartConfig"
 import { DiscreteBarDatum } from "./DiscreteBarChart"
-import { IChartTransform } from "./IChartTransform"
+import { ChartTransform } from "./ChartTransform"
 import { DimensionWithData } from "./DimensionWithData"
 import { ColorSchemes } from "./ColorSchemes"
 import { TickFormattingOptions } from "./TickFormattingOptions"
+import { Time } from "./TimeBounds"
 
 // Responsible for translating chart configuration into the form
 // of a discrete bar chart
-export class DiscreteBarTransform implements IChartTransform {
-    chart: ChartConfig
-
-    constructor(chart: ChartConfig) {
-        this.chart = chart
-    }
-
+export class DiscreteBarTransform extends ChartTransform {
```

Comment:
Nice. I think for core things like transforms using inheritance makes sense. (I was a little surprised that React discourages inheritance use so much, but at least for things other than components we could benefit from using it more).

Reply:
Now that I think about it, classes don't compose well, and we will likely eventually move away from using them in transforms.

Maybe a good example is that I didn't extend this class for `MapData` and `DataTableTransform` because the interfaces they provide are a little bit different. But they could still maybe share some subset of logic in the future.